# Individual Planning Report  
github link：https://github.com/zhifou541-stack/26-Elisha-Liu-Individual-Plan.git
# importing data

In [14]:
library(tidyverse)
url1='https://drive.google.com/uc?export=download&id=1LGq5BTBDuA8hTXNmLyc1A0I6Q0ifIL6e'
players<-read_csv(url1)
head(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [13]:
url2='https://drive.google.com/uc?export=download&id=1fJhos0VM3G2xg_0-oj48cyIv8rEs6udm'
sessions<-read_csv(url2)
head(sessions)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

1.Data Description:

The dataset consists of two files: " players.csv " and "session.csv".

In [15]:
nrow(players)
nrow(sessions)

[1] 196

[1] 1535

In the file players.csv, there are 196 observations and 7 variables in this file.

Among these variables, two are quantitative: played_hours and Age.
After calculating summary statistics, the average play time across all players is 5.42 hours, and the average player age is 19.37 years.

In [1]:
summary_table <- data.frame(Variable = c("played_hours", "Age"), Mean = c(5.42, 19.37))
summary_table

Variable     Mean 
1 played_hours  5.42
2 Age          19.37

These seven variables are:

+ **Experience:** character variable. It shows the players experience level. The categories — “Amateur,” “Regular,” “Veteran,” and “Pro” — indicate increasing levels of game proficiency in that order.

+ **subscribe:** logical variable. It indicates whether a player subscribed to the game.'True' means the player did subscribe the game, 'False' means 
the player did not subscribe the game.

+ **hashedEmail:** character variable. It is a hashed(anonnymized) version of the player's email.

+ **played_hours :** numeric variable. It shows the number of hours the player has spent on the game.

+ **name:** character variable. It is the display name chosen by each player in the game.

+ **gender:** character variable. It shows the gender of each player.

+ **Age：** numeric variable. It shows the player's age.

Based on the data, we can also see that subscription status seems to have some relationship with play time. Although players marked as “TRUE” don’t always play a lot — some even have just 0.1 hours — players marked as “FALSE” tend to play almost 0 hours overall.

在名为players.csv的file中：

there are 1535 observations and 5 variables in this file. and 这其中有3个quantitative variables，分别是start_time	end_time，original_start_time和original_end_time

这五个variable他们分别是：

+ **hashedEmail:** character variable, it is a hashed(anonnymized) version of the played's email.

+ **start_time:** character variable, it represents a play session's start time in “dd/mm/yyyy hh:mm” format.

+ **end_time:** character variable, it represents a play session's end timein “dd/mm/yyyy hh:mm” format.

+ **original_start_time:** numeric variable, it represents the precise session start time in system format.

+ **original_end_time:** numeric variable, it represents the precise session end time in system format.

2.Questions:

Board Question:

What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

Specific Question:


3.Exploratory Data Analysis and Visualization

4.Methods and Plan

5.GitHub Repository